### ISSUE TITLE
BCELoss in new version of SMP

### ISSUE THREAD

**Issue by SixtyTrees** [qubvel/segmentation_models.pytorch#773]
Hi,
I have a binary segmentation model that I trained on imbalanced data (1-2% of pixels belong to a class). On SMP 0.2.1 I've used

loss = smp.utils.losses.BCELoss() + smp.utils.losses.DiceLoss()
metrics = [smp.utils.metrics.IoU(threshold=threshold)]

And 

train_epoch = smp.utils.train.TrainEpoch(model, loss=loss, metrics=metrics, optimizer=optimizer, device=DEVICE, verbose=True, )
valid_epoch = smp.utils.train.ValidEpoch( model,  loss=loss, metrics=metrics, device=DEVICE,  verbose=True,)

Is there an analog of BCELoss? I only found soft_bce - does it act the same as soft_bce?
How to rewrite this code for current version?

I consider retraining models using current version of SMP, because the older version might become incompatible with other modern libraries.
Thank you

**Comment by SixtyTrees** [comment]
Thank you for your prompt response. I will try. And fall back to earlier version of SMP in the worst case.

Offtopic. One more question. Is there a good resource on neural networks, that doesn't just talk about "neuron is this" and "loss function does that", but rather has Python scripts to fully ran a training on an advanced framework? So far I ran into situations like these. Find the code, try to run it. Get an obfuscated error. Spend days googling it. What is your best practce to study and implement a new feature in the CNN field?

**Comment by qubvel** [comment]
You could start with documentation for Pytorch and Pytorch-Lightning, there are plenty of training examples there
